### DISCUSSION TITLE
Where does prisma store its data around what migrations have been applied to the database?

### DISCUSSION THREAD

**Discussion by josefaidt** [prisma/prisma#18434] [Q&A] [answered]
Title pretty much says it all. How does prisma know if there is a migration that has not yet been applied? If I run migrations on another machine and don't commit back to git, am I safe for subsequent migrations?

**Comment by ludralph** [comment, marked as answer]
Hi @josefaidt 👋 

Prisma stores information about applied database migrations in a table called `_prisma_migrations`. This table is created automatically when you run migrations using the prisma migrate command.

The `_prisma_migrations` table has a record for each applied migration, which includes information such as the migration name, the timestamp when it was applied, and the SHA256 hash of the migration script. This information is used by Prisma to determine which migrations have already been applied to the database, and which ones still need to be applied.

Prisma knows if there is a migration that has not yet been applied by comparing the migration history stored in the `_prisma_migrations` table in the database with the migration history stored in the Prisma migration folder in your project.

When you run migrations on a machine and apply them to the database, the migration history is stored in the `_prisma_migrations` table. If you don't commit the migration files to Git, subsequent migrations on another machine will not be aware of the new migrations that were applied on the previous machine.

To ensure that all machines are aware of the applied migrations, you should commit the migration files to Git and make sure that all developers pull the latest changes before running prisma migrate on their machines. This will ensure that the migration history is consistent across all machines. You can learn more from the migrate page in the [documentation](https://www.prisma.io/docs/concepts/components/prisma-migrate)

**Reply by josefaidt** [reply]
Hey @ludralph thank you for the in-depth response! This is extremely helpful and settles my concerns. Thank you again!

**Reply by King-witcher** [reply]
Migrations doesn't seem to be stored in that table. Of course Prisma stores something there, but I noticed that even if I drop that table, prisma still knows which migrations has been run or not.

**Comment by nikkhn** [comment]
@ludralph follow up question to this, lest say there's a discrepancy between the actual database schema and the migrations (e.g. you drop an entire table from the database, but want to re-build it with migrations). Since the `_prisma_migrations` table will already show the migration that created the table in the first place to already have been applied, is there a way to safely re-create the dropped table with migrations? or is it best to re-create the table manually? or use something like `prisma migrate diff` ?
